.github/workflows/docker-image.yml

```diff
@@ -8,21 +7,31 @@ on:
     paths-ignore: [ 'README.md']
 
 jobs:
-
-  build:
-
+  job-one:
+    name: job-one
     runs-on: ubuntu-latest
-
     steps:
-    - uses: actions/checkout@v3
-    - name: Build the Docker image
-      run: docker build --target test --tag my-test-image .
-    - run: docker run --env-file .env.test my-test-image tests   
-    - run: docker run -e TRELLO_KEY=${{ secrets.TRELLO_KEY }} -e TOKEN=${{ secrets.TOKEN }} my-test-image tests_e2e
-
-
-
-
-
-
-
+      - uses: actions/checkout@v3
+      - name: Build the Docker image
+        run: docker build --target test --tag my-test-image .
+      - run: docker run --env-file .env.test my-test-image tests   
+      - run: docker run -e TRELLO_KEY=${{ secrets.TRELLO_KEY }} -e TOKEN=${{ secrets.TOKEN }} my-test-image tests_e2e
+
+  job-two:
+    name: job-two
+    runs-on: ubuntu-latest
+    needs: job-one
+    if: github.event_name == 'push' && github.ref_name == 'main'
+    
+    env: 
+      HEROKU_API_KEY: ${{ secrets.HEROKU_API_KEY }}
+      
+    steps:
+      - uses: actions/checkout@v3
+      - run: docker login -u jackt24 -p ${{ secrets.DOCKER_PASSWORD }}
```

Comment:
Minor, but we should ideally action this step using the `--password-stdin` flag rather than passing the password explicitly, to avoid leaking those credentials to shell logs. [See the docs](https://docs.docker.com/engine/reference/commandline/login/#provide-a-password-using-stdin) for more info, and an example of how that might look below:
```
echo ${{ secrets.DOCKER_PASSWORD }} | docker login -u jackt24 --password-stdin
```